src/sentry/taskworker/registry.py

```diff
@@ -126,19 +126,38 @@ def wrapped(func: Callable[P, R]) -> Task[P, R]:
         return wrapped
 
     def send_task(self, activation: TaskActivation, wait_for_delivery: bool = False) -> None:
-        metrics.incr("taskworker.registry.send_task", tags={"namespace": activation.namespace})
-
         topic = self.router.route_namespace(self.name)
         produce_future = self._producer(topic).produce(
             ArroyoTopic(name=topic.value),
             KafkaPayload(key=None, value=activation.SerializeToString(), headers=[]),
         )
+        try:
+            if produce_future.exception(timeout=1):
+                metrics.incr(
+                    "taskworker.registry.send_task.failed",
+                    tags={"namespace": activation.namespace, "task_name": activation.taskname},
```

Comment:
```suggestion
                    tags={"namespace": activation.namespace, "taskname": activation.taskname},
```

Can we align on `taskname` as the metric tag name?